java-checks/src/main/java/org/sonar/java/checks/xml/spring/DefaultMessageListenerContainerCheck.java

```diff
@@ -0,0 +1,77 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.xml.spring;
+
+import com.google.common.collect.Iterables;
+import java.util.stream.IntStream;
+import java.util.stream.StreamSupport;
+import javax.xml.xpath.XPathExpression;
+import org.sonar.check.Rule;
+import org.sonar.java.xml.XPathXmlCheck;
+import org.sonar.java.xml.XmlCheckContext;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+
+@Rule(key = "S3439")
+public class DefaultMessageListenerContainerCheck extends XPathXmlCheck {
+
+  private XPathExpression defaultMessageListenerContainerBeanExpression;
+  private XPathExpression acceptMessageWhileStoppingPropertyExpression;
+  private XPathExpression sessionTransactedPropertyExpression;
+  private XPathExpression valueExpression;
+
+  @Override
+  public void precompileXPathExpressions(XmlCheckContext context) {
+    defaultMessageListenerContainerBeanExpression = context.compile("beans/bean[@class='org.springframework.jms.listener.DefaultMessageListenerContainer']");
+    acceptMessageWhileStoppingPropertyExpression = context.compile("property[@name='acceptMessagesWhileStopping']");
+    sessionTransactedPropertyExpression = context.compile("property[@name='sessionTransacted']");
+    valueExpression = context.compile("value[text()='true']");
+  }
+
+  @Override
+  public void scanFileWithXPathExpressions(XmlCheckContext context) {
+    StreamSupport.stream(context.evaluateOnDocument(defaultMessageListenerContainerBeanExpression).spliterator(), false)
+      .filter(bean -> !hasAcceptMessagePropertyEnabled(context, bean) && hasSessionTransactedDisabled(context, bean))
+      .forEach(bean -> reportIssue(bean, "Enable \"acceptMessagesWhileStopping\"."));
+  }
+
+  private boolean hasAcceptMessagePropertyEnabled(XmlCheckContext context, Node bean) {
+    return hasAttributeValue(bean, "acceptMessagesWhileStopping")
+      || hasPropertyAsChild(context, bean, acceptMessageWhileStoppingPropertyExpression);
+  }
+
+  private boolean hasSessionTransactedDisabled(XmlCheckContext context, Node bean) {
+    return !hasAttributeValue(bean, "sessionTransacted")
+      && !hasPropertyAsChild(context, bean, sessionTransactedPropertyExpression);
+  }
+
+  private static boolean hasAttributeValue(Node bean, String attributeName) {
+    NamedNodeMap attributes = bean.getAttributes();
+    return IntStream.range(0, attributes.getLength())
+      .mapToObj(attributes::item)
+      // ignore namespace
+      .anyMatch(attribute -> attribute.getNodeName().endsWith(attributeName) && "true".equals(attribute.getNodeValue()));
```

Comment:
My understanding of org.w3c.dom.Node API is for the following attribute 'acceptMessagesWhileStopping':
```xml
<beans xmlns:x="http://www.springframework.org/schema/p">
    <bean class="org.springframework.jms.listener.DefaultMessageListenerContainer"
        x:acceptMessagesWhileStopping="true"></bean>
</beans>
```
```
attribute.getPrefix()       -> "x"
attribute.getLocalName()    -> "acceptMessagesWhileStopping"
attribute.getNodeName()     -> "x:acceptMessagesWhileStopping"
attribute.getNamespaceURI() -> "http://www.springframework.org/schema/p"
```
So you should replace `attribute.getNodeName().endsWith(attributeName)` by `attributeName.equals(attribute.getLocalName())`

Reply:
Ok, let's keep the "endsWith" workaround, the risk to have FP/FN is near zero.